Tms/merchants endpoint #1

tmikeschu opened the pull request:
Worked with @Dpalazzari 

-Adds API endpoints for merchants: index, show, find, find_all, and random.

Needs in-depth review.

@sallymacnicholas we are having problems figuring out how to handle a string time parameter for finding merchants by created/updated_at. We've looked into this problem and can't figure it out. We've tried making a Time object from the string, and AR reads does not read them as the same. Thoughts?

Closes #1

tmikeschu committed:
Add rspec and factory girl setup

tmikeschu committed:
Add record endpoints for merchants

Index, show, find, find_all, and random.

Dpalazzari commented:
The functionality for merchants works, barring a few set backs. Those will be ironed out with conversation with teachers. Can I merge @tmikeschu ?

tmikeschu committed:
Refactor controller and route structure

Namespace for merchants and new controller folder

Dpalazzari commented on config/routes.rb:
```diff
@@ -1,9 +1,12 @@
+      namespace :merchants do
+        get '/find', to: 'search#show'
+        get '/find_all', to: 'search#index'
+        get '/random', to: 'random#show'
```
I like this. I think it's better and cleaner. @tmikeschu  👍 🚀

tmikeschu replied:
@Dpalazzari your idea! Sorry we didn't run with it to begin with.

tmikeschu commented:
@Dpalazzari , I agree that this basically works, but I'd feel more comfortable having a passing test for finding records by created/updated_at before merging. Thoughts?

Dpalazzari commented:
I agree. to be determined with conversation with @jmejia and @sallymacnicholas

@tmikeschu I might need this merged do begin work tomorrow morning though. Can we make a waffle card for the created?update _at tests so we can merge without forgetting to come back to these errors?

tmikeschu commented:
@Dpalazzari I noticed the same thing! Check the new PR that will allow testing for new endpoints. I'm going to follow the folder and namespacing in this branch structure as I work on a new one.

tmikeschu committed:
Merge branch 'master' of github.com:tmikeschu/rails_engine into tms/merchants-endpoint-#1

tmikeschu committed:
Change timestamps precision to 0

tmikeschu committed:
Refactor random show action

tmikeschu committed:
Add passing test for find by created at

tmikeschu committed:
Change name to case ignored text type

tmikeschu committed:
Add request helper method for attribute check

tmikeschu committed:
Add passing test for find by name case ignored

tmikeschu committed:
Add request tests

Tests for updated_at and finding by name ignoring case

tmikeschu committed:
Add seed for testing random merchant

tmikeschu commented:
@Dpalazzari : To review what we talked about: 

- Changed the name type to be 'citext', which is case-ignored by default.

- Removed the timestamp attributes and added them again (was unable to `change_column`) with `precision: 0` (see code). This makes it so the postgres timestamp and the json-ized timestamp are read as the same.

- Added request helper and refactored tests with a verify attributes method.

- Added and updated tests that involve case-insensitivity and timestamps.

Ready for merge!

@sallymacnicholas

tmikeschu committed:
Remove default pending model test

Dpalazzari commented:
Looks good @tmikeschu I will merge. 

FYI @jmejia @sallymacnicholas @case-eee
